src/wallet/wallet.cpp

```diff
@@ -1564,13 +1565,18 @@ CBlockIndex* CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, bool f
     CBlockIndex* pindex = pindexStart;
     CBlockIndex* ret = nullptr;
     {
-        LOCK2(cs_main, cs_wallet);
         fAbortRescan = false;
         fScanningWallet = true;
 
         ShowProgress(_("Rescanning..."), 0); // show rescan progress in GUI as dialog or on splashscreen, if -rescan on startup
-        double dProgressStart = GuessVerificationProgress(chainParams.TxData(), pindex);
-        double dProgressTip = GuessVerificationProgress(chainParams.TxData(), chainActive.Tip());
+        double dProgressStart = 0;
```

Comment:
Only `chainActive.Tip()` needs the lock? So keep old code and:
```cpp
CBlockIndex* tip = nullptr;
{
    LOCK(cs_main);
    tip = chainActive.Tip();
}

double dProgressStart = GuessVerificationProgress(chainParams.TxData(), pindex);
double dProgressTip = GuessVerificationProgress(chainParams.TxData(), tip);
```

Reply:
BTW, this patch allows the tip to be update so `dProgressTip` should be inside the loop?